db/db_impl/db_impl.cc

```diff
@@ -685,9 +685,18 @@ void DBImpl::StartTimedTasks() {
     stats_dump_period_sec = mutable_db_options_.stats_dump_period_sec;
     if (stats_dump_period_sec > 0) {
       if (!thread_dump_stats_) {
+        // In case of many `DB::Open()` in rapid succession we can have all
+        // threads dumping at once, which causes severe lock contention in
+        // jemalloc. Ensure successive `DB::Open()`s are staggered by at least
+        // one second in the common case.
+        static unsigned int stats_dump_threads_started = 0;
         thread_dump_stats_.reset(new ROCKSDB_NAMESPACE::RepeatableThread(
             [this]() { DBImpl::DumpStats(); }, "dump_st", env_,
-            static_cast<uint64_t>(stats_dump_period_sec) * kMicrosInSecond));
+            static_cast<uint64_t>(stats_dump_period_sec) * kMicrosInSecond,
+            static_cast<uint64_t>(stats_dump_threads_started %
```

Comment:
It may be rare case:
since `stats_dump_threads_started` has process life scope, using an `unsigned int` as a counter sounds a little scary to me. How about make it an `uint64_t`?

Reply:
Sure. I don't think unsigned int overflow is dangerous in this situation, but we can avoid it anyways since it's confusing.